src/tests/ftest/io/dfuse_bash_cmd.py

```diff
@@ -0,0 +1,212 @@
+
+           Args:
+             count(int): container index
+        """
+
+        # Get Dfuse params
+        self.dfuse = Dfuse(self.hostlist_clients[:-1], self.tmp)
+        self.dfuse.get_params(self)
+
+        # update dfuse params
+        self.dfuse.mount_dir.update("/tmp/" + self.pool.uuid + "_daos_dfuse"
+                                    + str(count))
+        self.dfuse.set_dfuse_params(self.pool)
+        self.dfuse.set_dfuse_cont_param(self.container)
+        self.dfuse.set_dfuse_exports(self.server_managers[0], self.client_log)
+
+        try:
+            # start dfuse
+            self.dfuse.run()
+        except CommandFailure as error:
+            self.log.error("Dfuse command %s failed on hosts %s",
+                           str(self.dfuse),
+                           str(NodeSet.fromlist(self.dfuse.hosts)),
+                           exc_info=error)
+            self.fail("Test was expected to pass but it failed.\n")
+
+    def test_bashcmd(self):
+        """Jira ID: DAOS-3508.
+
+        Test Description:
+            Purpose of this test is to mount different mount points of dfuse
+            for different container and pool sizes and perform basic bash
+            commands.
+        Use cases:
+            Folloing list of bash commands have been incorporated
+            as part of this test: mkdir, touch, ls, chmod, rm, dd, stat,
+            cp, cmp, mv, rmdir.
+              Create a directory.
+              Create a file under that directory.
+              List the created file.
+              Remove the file.
+              Write a file to the dfuse mounted location using dd.
+              List the written file to verify if it's create.
+              Verify the file created is of right size as desired.
+              Copy the file
+              Compare the copied file with original to verify the
+              content is same.
+              Remove copied file.
+              Rename file
+              Verify renamed file exist using list.
+              Remove a directory
+        :avocado: tags=all,hw,daosio,medium,ib2,full_regression,bashcmd
+        """
+        self.cont_count = self.params.get("cont_count", '/run/container/*')
+        self.pool_count = self.params.get("pool_count", '/run/pool/*')
+
+        # Create a pool if one does not already exist.
+        for _ in range(self.pool_count):
+            self.create_pool()
+            self.create_cont()
+            # perform test for multiple containers.
+            for count in range(self.cont_count):
+                self.start_dfuse(count)
+                abs_dir_path = os.path.join(
+                    self.dfuse.mount_dir.value, self.dir_name)
+                abs_file_path1 = os.path.join(abs_dir_path, self.file_name1)
+                abs_file_path2 = os.path.join(abs_dir_path, self.file_name2)
+                # check if the dir exists.
+                dir_exists, _ = general_utils.check_file_exists(
```

Comment:
Incorrect use of hostlist_clients again, but the dfuse object itself will perform this check now.